source/common/common/logger.cc

```diff
@@ -19,18 +19,32 @@ namespace Logger {
 StandardLogger::StandardLogger(const std::string& name)
     : Logger(std::make_shared<spdlog::logger>(name, Registry::getSink())) {}
 
-SinkDelegate::SinkDelegate(DelegatingLogSinkSharedPtr log_sink)
-    : previous_delegate_(log_sink->delegate()), log_sink_(log_sink) {
-  log_sink->setDelegate(this);
-}
+SinkDelegate::SinkDelegate(DelegatingLogSinkSharedPtr log_sink) : log_sink_(log_sink) {}
 
 SinkDelegate::~SinkDelegate() {
```

Comment:
Backing up a step, is all of this only needed for log overrides in test?  If so I wonder if a different injection model would allow us to avoid all the locks in production code.

Reply:
It's not only for tests. This is how the file access logger works also. With that said, I think it might be possible to somehow design this without locks, but it's not immediately clear to me how, so my preference would be to merge this as it's definitely broken and maybe put in a TODO to do something better? wdyt?

Reply:
Gotcha, bummer.  Yeah, sounds like uncontested locks is the best workaround for now.